src/control/lib/daos/container_property.go

```diff
@@ -0,0 +1,1214 @@
+*/
+import "C"
+
+// ContainerPropType defines a native Go type for the container property types
+// defined in the DAOS API.
+type ContainerPropType C.uint
+
+const (
+	containerPropMin             ContainerPropType = C.DAOS_PROP_CO_MIN
+	ContainerPropLabel           ContainerPropType = C.DAOS_PROP_CO_LABEL
+	ContainerPropLayout          ContainerPropType = C.DAOS_PROP_CO_LAYOUT_TYPE
+	ContainerPropLayoutVersion   ContainerPropType = C.DAOS_PROP_CO_LAYOUT_VER
+	ContainerPropChecksumEnabled ContainerPropType = C.DAOS_PROP_CO_CSUM
+	ContainerPropChecksumSize    ContainerPropType = C.DAOS_PROP_CO_CSUM_CHUNK_SIZE
+	ContainerPropChecksumSrvVrfy ContainerPropType = C.DAOS_PROP_CO_CSUM_SERVER_VERIFY
+	ContainerPropRedunFactor     ContainerPropType = C.DAOS_PROP_CO_REDUN_FAC
+	ContainerPropRedunLevel      ContainerPropType = C.DAOS_PROP_CO_REDUN_LVL
+	ContainerPropMaxSnapshots    ContainerPropType = C.DAOS_PROP_CO_SNAPSHOT_MAX
+	ContainerPropACL             ContainerPropType = C.DAOS_PROP_CO_ACL
+	ContainerPropCompression     ContainerPropType = C.DAOS_PROP_CO_COMPRESS
+	ContainerPropEncryption      ContainerPropType = C.DAOS_PROP_CO_ENCRYPT
+	ContainerPropOwner           ContainerPropType = C.DAOS_PROP_CO_OWNER
+	ContainerPropGroup           ContainerPropType = C.DAOS_PROP_CO_OWNER_GROUP
+	ContainerPropDedupEnabled    ContainerPropType = C.DAOS_PROP_CO_DEDUP
+	ContainerPropDedupThreshold  ContainerPropType = C.DAOS_PROP_CO_DEDUP_THRESHOLD
+	ContainerPropRootObjects     ContainerPropType = C.DAOS_PROP_CO_ROOTS
+	ContainerPropStatus          ContainerPropType = C.DAOS_PROP_CO_STATUS
+	ContainerPropHighestOid      ContainerPropType = C.DAOS_PROP_CO_ALLOCED_OID
+	ContainerPropEcCellSize      ContainerPropType = C.DAOS_PROP_CO_EC_CELL_SZ
+	ContainerPropEcPerfDom       ContainerPropType = C.DAOS_PROP_CO_EC_PDA
+	ContainerPropEcPerfDomAff    ContainerPropType = C.DAOS_PROP_CO_RP_PDA
+	ContainerPropGlobalVersion   ContainerPropType = C.DAOS_PROP_CO_GLOBAL_VERSION
+	ContainerPropScubberDisabled ContainerPropType = C.DAOS_PROP_CO_SCRUBBER_DISABLED
+	ContainerPropObjectVersion   ContainerPropType = C.DAOS_PROP_CO_OBJ_VERSION
+	ContainerPropPerfDomain      ContainerPropType = C.DAOS_PROP_CO_PERF_DOMAIN
+	containerPropMax             ContainerPropType = C.DAOS_PROP_CO_MAX
+)
+
+func (cpt ContainerPropType) String() string {
+	switch cpt {
+	case ContainerPropLabel:
+		return C.DAOS_PROP_ENTRY_LABEL
+	case ContainerPropLayout:
+		return C.DAOS_PROP_ENTRY_LAYOUT_TYPE
+	case ContainerPropLayoutVersion:
+		return C.DAOS_PROP_ENTRY_LAYOUT_VER
+	case ContainerPropChecksumEnabled:
+		return C.DAOS_PROP_ENTRY_CKSUM
+	case ContainerPropChecksumSize:
+		return C.DAOS_PROP_ENTRY_CKSUM_SIZE
+	case ContainerPropChecksumSrvVrfy:
+		return C.DAOS_PROP_ENTRY_SRV_CKSUM
+	case ContainerPropRedunFactor:
+		return C.DAOS_PROP_ENTRY_REDUN_FAC
+	case ContainerPropRedunLevel:
+		return C.DAOS_PROP_ENTRY_REDUN_LVL
+	case ContainerPropMaxSnapshots:
```

Comment:
I would be more comfortable if the go constant names matched the C constant names exactly or at least contained the same strings in the same order even if we expand the abbreviations.

Reply:
Renamed ContainerPropLayout -> ContainerPropLayoutType